### PR TITLE
Visual Update for computeparticles

### PR DESCRIPTION
**Pull request by Voultapher**
Instead of using a small amount of large textured particles, use a large
amount of small monochrome particles.
Which uses a new vulkanexamplebase functionality of creating and
updating a only device visible buffer via a temporary staging buffer.

### CODE REVIEW

**Comment by SaschaWillems** [comment]
Thx for this updated PR, I'll try to get to it after work :+1:

File: computeparticles/computeparticles.cpp [hunk 1]
```diff
@@ -504,16 +378,16 @@ class VulkanExample : public VulkanExampleBase
...
+				VK_FALSE,
+				VK_COMPARE_OP_ALWAYS);
 
 		VkPipelineViewportStateCreateInfo viewportState =
 			vkTools::initializers::pipelineViewportStateCreateInfo(1, 1, 0);
 
 		VkPipelineMultisampleStateCreateInfo multisampleState =
 			vkTools::initializers::pipelineMultisampleStateCreateInfo(
-				VK_SAMPLE_COUNT_1_BIT,
+				VK_SAMPLE_COUNT_4_BIT,
```
Comment (SaschaWillems) [inline computeparticles/computeparticles.cpp]: Any reason for this? Unless multi sampling attachments are created and used, upping the sample count here won't make any sense :wink:

Comment (Voultapher) [inline computeparticles/computeparticles.cpp]: My bad, another one of those relics.

Comment (SaschaWillems) [inline computeparticles/computeparticles.cpp]: No problem, I'll just revert that after I merged. Just wanted to let you know.

**Pull request merged**